file-formats/gsmp/type/zif_aff_gsmp_v1.intf.abap

```diff
@@ -0,0 +1,179 @@
+INTERFACE zif_aff_gsmp_v1
+  PUBLIC.
+
+  TYPES:
+
+    "! <p class="shorttext">Provider Type</p>
+    "! Provider type
+    "! $values {@link zif_aff_gsmp_v1.data:co_provider_type}
+    "! $default {@link zif_aff_gsmp_v1.data:co_provider_type.class}
+    ty_provider_type  TYPE c LENGTH 4,
+
+    "! <p class="shorttext">Execution Mode</p>
+    "! Execution mode
+    "! $values {@link zif_aff_gsmp_v1.data:co_execution_mode}
+    ty_execution_mode TYPE c LENGTH 1,
+
+
+    "! <p class="shorttext">Provider Position</p>
+    "! Provider position
+    "! $values {@link zif_aff_gsmp_v1.data:co_provider_pos}
+    "! $default {@link zif_aff_gsmp_v1.data:co_provider_pos.normal}
+    ty_provider_pos   TYPE i,
+
+
+    "! <p class="shorttext">Provider Implementation</p>
+    "! Provider implementation
+    BEGIN OF ty_provider_implementation,
+
+      "! <p class="shorttext">Type</p>
+      "! Type
+      "! $required
+      type TYPE ty_provider_type,
+
+      "! <p class="shorttext">Name</p>
+      "! Name
+      "! $required
+      name TYPE zif_aff_types_v1=>ty_object_name_30,
+
+    END OF ty_provider_implementation,
+
+
+
+    "! <p class="shorttext">Provider Execution</p>
+    "! Provider execution
+    BEGIN OF ty_provider_execution,
+
+      "! <p class="shorttext">Execution Mode</p>
+      "! Execution mode
+      "! $required
+      execution_mode TYPE ty_execution_mode,
+
+      "! <p class="shorttext">Provider Position</p>
+      "! Provider Position
+      "! $required
+      position   TYPE ty_provider_pos,
+
+      "! <p class="shorttext">Scope Dependent</p>
+      "! Scope dependent
+      "! $required
+      scope_dependent       TYPE abap_bool,
+
+    END OF ty_provider_execution,
+
+
+
+    "! <p class="shorttext">Provider Data Model</p>
+    "! Provider data model
+    BEGIN OF ty_data_model,
+
+      "! <p class="shorttext">JSON Data Model</p>
+      "! JSON data model
+      model_data     TYPE string,
```

Comment:
I am struggling with the field `model_data` in the structure `data_model`. Which data do we expect in this field. Is it JSON source code? if so, would it make sense to move it to a dedicated file?

Reply:
Before the last change the field was called `model_data`. If it shall be a dedicated field (no additional file), I can think of a structure `model` with a field `data`. What do you think?